File: hanoi/hanoi.c
#include <stdio.h>
#include <stdlib.h>

void move(char, char);
void hanoi(int, char, char, char);

int main(int argc, const char **argv)
{
    if (argc < 2) {
        return 1;
    } else {
        hanoi(atoi(argv[1]), 'A', 'C', 'B');
        return 0;
    }
}

void move(char from, char to)
{
   printf("Move a disc from %c to %c.\n", from, to);
} 


void hanoi(int n, char from, char to, char via)
{
    if (n == 0) {
        return;
    } else {
        hanoi(n - 1, from, via, to);
        move(from, to);
        hanoi(n - 1, via, to, from);
    }
}

File: hanoi/hanoi.py
def move_disc(_from, _to):
    print(f'{_from} to {_to}')

def hanoi(_n, _from, _to, _via):
    if _n == 0:
        pass
    else:
        hanoi(_n - 1, _from, _via, _to)
        move_disc(_from, _to)
        hanoi(_n - 1, _via, _to, _from)

def main(argv):
    hanoi(int(argv[1]), 'A', 'C', 'B')

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

main(['a', '3'])
